mlir/test/Dialect/SCF/transform-ops.mlir

```diff
@@ -168,6 +168,164 @@ module attributes {transform.with_named_sequence} {
+}
+
+module attributes {transform.with_named_sequence} {
+  transform.named_sequence @__transform_main(%arg1: !transform.any_op {transform.readonly}) {
+    %0 = transform.structured.match ops{["memref.store"]} in %arg1 : (!transform.any_op) -> !transform.any_op
+    %1 = transform.get_parent_op %0 {op_name = "scf.for"} : (!transform.any_op) -> !transform.op<"scf.for">
+    transform.loop.unroll_and_jam %1 { factor = 4 } : !transform.op<"scf.for">
+    transform.yield
+  }
+}
+
+// -----
+
+// CHECK-LABEL: @loop_unroll_and_jam_op
+// CHECK:  %[[VAL_0:.*]]: memref<21x30xf32, 1>, %[[VAL_1:.*]]: f32, %[[VAL_2:.*]]: f32) {
+func.func @loop_unroll_and_jam_op(%arg0: memref<21x30xf32, 1>, %init : f32, %init1 : f32) {
+  // CHECK:           %[[VAL_3:.*]] = arith.constant 20 : index
+  // CHECK:           %[[VAL_4:.*]]:2 = affine.for %[[VAL_5:.*]] = 0 to 20 step 2 iter_args(%[[VAL_6:.*]] = %[[VAL_1]], %[[VAL_7:.*]] = %[[VAL_1]]) -> (f32, f32) {
+  %0 = affine.for %arg3 = 0 to 21 iter_args(%arg4 = %init) -> (f32) {
+    // CHECK:             %[[VAL_8:.*]]:2 = affine.for %[[VAL_9:.*]] = 0 to 30 iter_args(%[[VAL_10:.*]] = %[[VAL_2]], %[[VAL_11:.*]] = %[[VAL_2]]) -> (f32, f32) {
+    %1 = affine.for %arg5 = 0 to 30 iter_args(%arg6 = %init1) -> (f32) {
+      // CHECK:               %[[VAL_12:.*]] = affine.load %[[VAL_0]]{{\[}}%[[VAL_5]], %[[VAL_9]]] : memref<21x30xf32, 1>
+      // CHECK:               %[[VAL_13:.*]] = arith.addf %[[VAL_10]], %[[VAL_12]] : f32
+      // CHECK:               %[[VAL_14:.*]] = affine.apply #map(%[[VAL_5]])
+      // CHECK:               %[[VAL_15:.*]] = affine.load %[[VAL_0]]{{\[}}%[[VAL_14]], %[[VAL_9]]] : memref<21x30xf32, 1>
+      // CHECK:               %[[VAL_16:.*]] = arith.addf %[[VAL_11]], %[[VAL_15]] : f32
+      // CHECK:               affine.yield %[[VAL_13]], %[[VAL_16]] : f32, f32
+      %3 = affine.load %arg0[%arg3, %arg5] : memref<21x30xf32, 1>
+      %4 = arith.addf %arg6, %3 : f32
+      affine.yield %4 : f32
+    }
+    // CHECK:             %[[VAL_17:.*]] = arith.mulf %[[VAL_6]], %[[VAL_18:.*]]#0 : f32
+    // CHECK:             %[[VAL_19:.*]] = affine.apply #map(%[[VAL_5]])
+    // CHECK:             %[[VAL_20:.*]] = arith.mulf %[[VAL_7]], %[[VAL_18]]#1 : f32
+    // CHECK:             affine.yield %[[VAL_17]], %[[VAL_20]] : f32, f32
+    // CHECK:           }
+    // CHECK:           %[[VAL_21:.*]] = arith.mulf %[[VAL_22:.*]]#0, %[[VAL_22]]#1 : f32
+    // CHECK:           %[[VAL_23:.*]] = affine.for %[[VAL_24:.*]] = 0 to 30 iter_args(%[[VAL_25:.*]] = %[[VAL_2]]) -> (f32) {
+    // CHECK:             %[[VAL_26:.*]] = affine.load %[[VAL_0]]{{\[}}%[[VAL_3]], %[[VAL_24]]] : memref<21x30xf32, 1>
+    // CHECK:             %[[VAL_27:.*]] = arith.addf %[[VAL_25]], %[[VAL_26]] : f32
+    // CHECK:             affine.yield %[[VAL_27]] : f32
+    %2 = arith.mulf %arg4, %1 : f32
+    affine.yield %2 : f32
+  }
+  // CHECK:           %[[VAL_28:.*]] = arith.mulf %[[VAL_21]], %[[VAL_29:.*]] : f32
+  return
+}
+
```

Comment:
Please add a test where the Factor is bigger than the trip count.

Reply:
Also check the case that having only yield op in the block will not change the IR.

Reply:
Ack